src/utils/makeReactNativeConfig.js

```diff
@@ -270,4 +366,35 @@ function injectPolyfillIntoEntry(
   return userEntry;
 }
 
-module.exports = { makeReactNativeConfig, injectPolyfillIntoEntry };
+function createWebpackConfig(configBuilder: WebpackConfigFactory) {
+  return (options: EnvOptions) => {
+    const haulWebpackConfiguration =
+      typeof configBuilder === 'function'
```

Comment:
Would it be okay to make one more check for `configBuilder`, if it's our required format? It's exposed to user directly and yet, we don't check if even `entry` is valid.

Reply:
I don't think it's necessary, as we don't handle errors in config itself as well (for easier debug on user's side) we should trust user that he knows what he doing.

cc @satya164 @zamotany

Reply:
Yes, it'll be good validate options passed to `createWebpackConfig`.

>  as we don't handle errors in config itself as well

yeah, but webpack will throw in that case, so it'll be duplicating the logic

> for easier debug on user's side) we should trust user that he knows what he doing

that was for `require` statement. when user returns their own webpack config, we should trust the user. but for arguments to our `createWebpackConfig`, if we don't validate it, it'll make debugging harder

Reply:
I got it that it was for `require`. But this is the same case – we should trust the user. 

```import { createWebpackConfig } from 'haul';

export default {
  webpack: env => {
    const config = createWebpackConfig({
      entry: `./index.js`,
      include: /node_modules/,
    })(env);

    config.plugins.push(new CaseSensitivePathsPlugin());

    return config;
  },
}```

Reply:
As I said, when user returns their own webpack config, we should trust the user. But @Krizzu is suggesting to validate arguments to `createWebpackConfig`, which is our own function. We know what are valid arguments and the user cannot pass arbitrary config here.

Reply:
> In this example, configBuilder is an anonymous function

Reply:
As I said, it has not to be, as I've shown you in the example literally one post above.

Reply:
So If I understand you correctly, you proposing to check if `haulWebpackConfiguration` has `entry` field? Of if user passing to `createWebpackConfig` object or function with arity 1 (we accept one argument - env object)?

Reply:
It might be a good idea to check if the object has keys what we are going to use - currently, we support only "entry". I can use Joi as object schema validator (https://github.com/hapijs/joi) to be prepared for rich cases. What do you think? @satya164

Reply:
```js

const config = typeof configBuilder === 'function' ? configBuilder(options) : configBuilder;

if (typeof config !== 'object' || config === null) {
  throw new Error(`The arguments passed to 'createWebpackConfig' must be an object or a function returning an object.`).
}

if (typeof config.entry !== 'string') {
  throw new Error(`The 'entry' property must be a string and point to your app's entry point (usually 'index.js').`);
}

if (!fs.existsSync(path.resolve(process.cwd(), config.entry))) {
  throw new Error(`The file '${config.entry}' doesn't exist. It should point to your app's entry point (usually 'index.js')`);
}
```

Reply:
Don't think we need any validation library right now. If it gets more complex in future, we'll add one.

Reply:
at Joi – ok, makes sense.